source/docs/software/commandbased/builtins.rst

```diff
@@ -0,0 +1,379 @@
+      :language: c++
+      :lines: 25-44
+      :linenos:
+      :lineno-start: 25
+
+ConditionalCommand
+^^^^^^^^^^^^^^^^^^
+
+``ConditionalCommand`` (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/ConditionalCommand.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classfrc2_1_1_conditional_command.html>`__) is a specialized version of this that decides between two commands using a boolean condition.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    // Runs either commandOnTrue or commandOnFalse depending on the value of m_limitSwitch.get()
+    new ConditionalCommand(commandOnTrue, commandOnFalse, m_limitSwitch::get)
+
+  .. code-tab:: c++
+
+    // Runs either commandOnTrue or commandOnFalse depending on the value of m_limitSwitch.get()
+    frc2::ConditionalCommand(commandOnTrue, commandOnFalse, [&m_limitSwitch] { return m_limitSwitch.Get(); })
+
+
+Waiting For Delays
+------------------
+
+Wait for a certain condition to happen or adding a delay can be useful to synchronize between different commands in a command group or between other robot actions.
+
+WaitCommand
+^^^^^^^^^^^
+
+``WaitCommand`` (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/WaitCommand.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classfrc2_1_1_wait_command.html>`__) does nothing and ends after a specified period of time elapses.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    // Ends 5 seconds after being scheduled
+    new WaitCommand(5.0)
+
+  .. code-tab:: c++
+
+    // Ends 5 seconds after being scheduled
+    frc2::WaitCommand(5.0_s)
+
+WaitUntilCommand
+^^^^^^^^^^^^^^^^
+
+``WaitUntilCommand`` (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/WaitUntilCommand.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classfrc2_1_1_wait_until_command.html>`__) does nothing, ending once a specified condition becomes true or once a specified match time passes.
+
+.. warning:: The match timer used by WaitUntilCommand does *not* provide an official match time! While it is fairly accurate, use of this timer can *not* guarantee the legality of your robot's actions.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    // Ends after the 60-second mark of the current match
+    new WaitUntilCommand(60.0)
+
+    // Ends after m_limitSwitch.get() returns true
+    new WaitUntilCommand(m_limitSwitch::get)
+
+  .. code-tab:: c++
+
+    // Ends after the 60-second mark of the current match
+    frc2::WaitUntilCommand(60.0_s)
+
+    // Ends after m_limitSwitch.Get() returns true
+    frc2::WaitUntilCommand([&m_limitSwitch] { return m_limitSwitch.Get(); })
+
+In combination with ``ParallelRaceGroup`` or ``ParallelDeadlineGroup``, this can be used to replace a command's end condition or add another one - in fact, that is what the ``until()`` decorator does under the hood.
```

Comment:
```suggestion
In combination with ``ParallelRaceGroup`` or ``ParallelDeadlineGroup``, this can be used to replace a command's end condition or add another one. In fact, that is what the ``until()`` decorator does under the hood.
```